### ISSUE TITLE
UriHelpers has no null checks, consider throwing if null

### ISSUE THREAD

**Issue by jkotalik** [aspnet/HttpAbstractions#682]
If a param is null, we null ref out rather than throwing an argument exception.
One example:

```
public static string BuildRelative(
            PathString pathBase = new PathString(),
            PathString path = new PathString(),
            QueryString query = new QueryString(),
            FragmentString fragment = new FragmentString())
        {
            string combinePath = (pathBase.HasValue || path.HasValue) ? (pathBase + path).ToString() : "/";
            return combinePath + query.ToString() + fragment.ToString();
        }
```

Though these have defaults, if a client passes in null, it will null ref out.

**Comment by dougbu** [comment]
Why do we have optional parameters in a public API? [Engineering guidelines](https://github.com/aspnet/Home/wiki/Engineering-guidelines#async-method-patterns) mention `default(CancellationToken)` is "one of the few places that we use optional parameters".

Probably can't fix this in 1.1.0 (because back-compat) but do we need a 2.0 issue about splitting this method into multiple overloads?

**Comment by jkotalik** [comment]
Too late to fix the splitting into multiple overloads for 2.0.0, but will add null checks for 2.1
